packages/cyberstorm/src/components/CopyButton/CopyButton.tsx

```diff
@@ -0,0 +1,72 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faClone } from "@fortawesome/free-regular-svg-icons";
+import { faCheck } from "@fortawesome/free-solid-svg-icons";
+import React, { PropsWithChildren, useEffect } from "react";
+import { useState } from "react";
+import styles from "./CopyButton.module.css";
+import { Tooltip } from "../Tooltip/Tooltip";
+
+type CopyButtonProps = {
+  text: string;
+};
+type CopiedValue = string | null;
+type CopyFn = (text: string) => Promise<void>;
+
+function useCopyToClipboard(): CopyFn {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [copiedText, setCopiedText] = useState<CopiedValue>(null);
+
+  const copy: CopyFn = async (text) => {
+    if (!navigator?.clipboard) {
+      console.warn("Clipboard not supported");
```

Comment:
If the clipboard can't be used, maybe we should just `return;` here after logging the warning?

Although it would be more user-friendly if the `CopyButton` would also do this check, and just return/render `null` if the copy functionality isn't available. Why show a button at all if clicking it only logs some errors in the console?